packages/in_app_purchase/lib/src/in_app_purchase/google_play_connection.dart

```diff
@@ -220,23 +245,18 @@ class GooglePlayConnection
   static Future<List<PurchaseDetails>> _getPurchaseDetailsFromResult(
       PurchasesResultWrapper resultWrapper) async {
     IAPError error;
-    PurchaseStatus status;
-    if (resultWrapper.responseCode == BillingResponse.ok) {
-      error = null;
-      status = PurchaseStatus.purchased;
```

Comment:
How is the status set now when the response is OK?

Reply:
`PurchaseDetails.fromPurchase` now sets the status since the purchaseState is not part of the `Purchase` object